src/file_lines.rs

```diff
@@ -0,0 +1,221 @@
+//! This module contains types and functions to support formatting specific line ranges.
+use std::{cmp, iter, str};
+
+use itertools::Itertools;
+use multimap::MultiMap;
+use rustc_serialize::{self, json};
+
+use codemap::LineRange;
+
+/// A range that is inclusive of both ends.
```

Comment:
My reasoning was: for a human: saying "please format lines 7-15" and finding that line 15 was not formatted would be confusing. The line numbers in the `Loc` that come back from `CodeMap::lookup_char_pos` are 1-based rather than 0-based which kind of supports this being unconventional: https://crates.fyi/crates/syntex_syntax/0.33.0/codemap/struct.Loc.html#structfield.line

I don't have strong feelings though, and can change this if you prefer.